### PR TITLE
feat:added wishlist management and signup auth

### PR DESCRIPTION
**Pull request by tushargoyal812**
https://user-images.githubusercontent.com/69681896/160590980-83dfcf79-94da-490a-8b55-ac27909632f2.mp4

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import {Home} from './pages/home/home'
 import {Products} from './pages/products/products'
 import {Cart} from './pages/cart/cart'
 import { Login } from "./pages/login/login";
+import { SignUp } from "./pages/sign-up/sign-up";
+import { Wishlist } from "./pages/wishlist/wishlist";
 import {Routes,Route} from 'react-router-dom'
```
Comment (sharmayatendra) [inline src/App.js:8]: you can use single index file to import the pages

File: src/components/nav/nav.js [hunk 2]
```diff
@@ -2,10 +2,12 @@ import './nav.css'
 import {Link} from 'react-router-dom'
 import { useCart } from '../../filter-context/cart-context'
 import { useAuth } from '../../filter-context/auth-context'
+import { useWishlist } from '../../filter-context/wishlist-context'
```
Comment (sharmayatendra) [inline src/components/nav/nav.js:5]: same here also can make single index file

File: src/App.js [hunk 3]
```diff
@@ -13,6 +15,8 @@ function App() {
         <Route path='/products' element={<Products/>}/>
         <Route path='/cart' element={<Cart/>}/>
         <Route path='/login' element={<Login/>}/>
+        <Route path='/signup' element={<SignUp/>}/>
+        <Route path='/wishlist' element={<Wishlist/>}/>
```
Comment (deepak29-git) [inline src/App.js:19]: add page not found route so that if user type wrong url then show the page not found

File: src/pages/cart/cart.js [hunk 4]
```diff
@@ -20,15 +25,51 @@ export const Cart=()=>{
...
+        if(wishlist.some(wishItem=>wishItem.id===product.id))
+        {
+            removeFromCartHandler(product)
+        }else{
+            try {
+                const response=await axios.post('/api/user/wishlist',{
+                    product
+                },{
+                  headers: {
+                    authorization: token, // passing token as an authorization header
```
Comment (deepak29-git) [inline src/pages/cart/cart.js:45]: remove comment

**Review by deepak29-git — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (tushargoyal812) [reply src/App.js:8]: ok will do it

File: src/components/nav/nav.js (hunk 2 above)
Comment (tushargoyal812) [reply src/components/nav/nav.js:5]: ok got it

File: src/App.js (hunk 3 above)
Comment (tushargoyal812) [reply src/App.js:19]: ok got it

File: src/pages/cart/cart.js (hunk 4 above)
Comment (tushargoyal812) [reply src/pages/cart/cart.js:45]: ok

**Review by sharmayatendra — APPROVED** [review]
give a few suggestion can look into it

1- good file structure
2- clean code

**Pull request merged**